spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractLocalDeployerSupport.java

```diff
@@ -213,11 +218,20 @@ protected void handleAppPropertiesPassing(AppDeploymentRequest request, Map<Stri
 											  Map<String, String> appPropertiesToUse) {
 		if (useSpringApplicationJson(request)) {
 			try {
+				//If SPRING_APPLICATION_JSON is found, explode it and merge back into appProperties
+				Map<String, String> localApplicationProperties = new HashMap<>(appProperties);
+				if(localApplicationProperties.containsKey(SPRING_APPLICATION_JSON)){
+					localApplicationProperties.putAll(OBJECT_MAPPER.readValue(appProperties.get(SPRING_APPLICATION_JSON), new TypeReference<HashMap<String,Object>>() {}));
+					localApplicationProperties.remove(SPRING_APPLICATION_JSON);
+				}
 				appInstanceEnvToUse.putAll(Collections.singletonMap(
-						"SPRING_APPLICATION_JSON", OBJECT_MAPPER.writeValueAsString(appProperties)));
+						SPRING_APPLICATION_JSON, OBJECT_MAPPER.writeValueAsString(localApplicationProperties)));
 			} catch (JsonProcessingException e) {
 				throw new RuntimeException(e);
 			}
+			catch (IOException e) {
+				throw new RuntimeException(e);
+			}
```

Comment:
Wrapping these two in just a `RuntimeException` feels a bit lax.  Should we create a custom exception for this?  Also, if this project requires java 8, we can use the multiple exception syntax:
```
catch(JsonProcessingException | IOException ex){
```

Reply:
Multiple exception syntax is a no brainer for sure, forgot this was a java8 project. Adding a new Exception, maybe at the `buildProcessBuilder`, would be nice if the `AppDeployer` had some sort of `DeploymentFailedException` so at least we could base out of that. Creating an exception to be caught at the LocalAppDeployer to propagate `RuntimeException` as it happens for other things, seems like that would not be the right place. Another example where we may need to touch the SPI.

@jvalkeal made a comment recently on how the k8s deployer needs some love, I'd say they all need some attention to get to a more cohesive state. They all obey the spec of course, but we could benefit from some patterns that happen on all different deployers (the EntryPoint style is one of many)

Reply:
BTW we can't have multiple catch syntax on this, because JsonProcessingException is a subclass of IOException.

Reply:
Then why have both catches in the first place?

Reply:
I'm fixing this, I'm not saying we need both either. This was just the IDE suggesting the inclusion of IOException, where we had `JsonProcessingException` already in place.